scripts/ccpp_datafile.py

```diff
@@ -438,21 +457,30 @@ def _is_variable_protected(table, var_name, var_dict):
             # end for
         # end if
         parent = var_dict.get("parent")
-        var_dict = _find_var_dictionary(table, parent)
+        if parent is not None:
+            var_dict = _find_var_dictionary(table, dict_name=parent)
+        else:
+            var_dict = None
+        # end if
     # end while
     return protected
 
 ###############################################################################
-def _retrieve_variable_list(table, suite_name, intent_type=None):
+def _retrieve_variable_list(table, suite_name,
+                            intent_type=None, excl_prot=True):
 ###############################################################################
     """Find and return a list of all the required variables in <suite_name>.
     If suite, <suite_name>, is not found in <table>, return an empty list.
     If <intent_type> is present, return only that variable type (input or
-    output)."""
+    output).
+    If <excl_prot> is True, do not include protected variables"""
```

Comment:
Can you explain why this functionality to exclude protected variables is/may be needed?

Reply:
We use some of the reports to find variables that will need initializing. We cannot read in values for `protected_variables` so they are (optionally) excluded.